gems/pending/spec/spec_helper.rb

```diff
@@ -32,3 +36,30 @@
   config.backtrace_exclusion_patterns << %r{/lib\d*/ruby/[0-9]}
   config.backtrace_exclusion_patterns << %r{/gems/[0-9][^/]+/gems/}
 end
+
+#
+# So tests can clear class-level caches between examples.
+# TODO: Add this to the azure-armrest gem.
+#
+class Azure::Armrest::ArmrestService
+  def self.clear_caches
+    @@providers_hash = {}
+    @@tokens         = {}
+    @@subscriptions  = {}
+  end
+end
```

Comment:
a) If you're adding this as actual functionality to the gem that you plan to add later, I'd probably move this out of this specs-only helper into a normal patch elsewhere. It doesn't really belong here.

b) If this is very much for the sake of specs, perhaps a more appropriate thing to do would be to load a set of [test macros](https://www.relishapp.com/rspec/rspec-core/docs/helper-methods/define-helper-methods-in-a-module) (methods loaded in to the examples) that do a simple `class_eval` to clear these rather than patching on a new method onto the class.

I believe either of these options would also allow you to remove the azure-armrest require here which isn't needed by the other specs utilizing this spec_helper.